x-pack/plugins/cloud_security_posture/server/types.ts

```diff
@@ -49,19 +46,22 @@ export interface CspServerPluginStartDeps {
 export type CspServerPluginStartServices = Promise<
   [CoreStart, CspServerPluginStartDeps, CspServerPluginStart]
 >;
-export type CspRequestHandlerContext = FleetRequestHandlerContext;
 
-/**
- * Convenience type for request handlers in CSP that includes the CspRequestHandlerContext type
- * @internal
- */
-export type CspRequestHandler<
-  P = unknown,
-  Q = unknown,
-  B = unknown,
-  Method extends RouteMethod = any,
-  ResponseFactory extends KibanaResponseFactory = KibanaResponseFactory
-> = RequestHandler<P, Q, B, CspRequestHandlerContext, Method, ResponseFactory>;
+interface CspApiRequestHandlerContext {
+  logger: Logger;
+  security: SecurityPluginStart;
+  fleet: {
+    agentPolicyService: AgentPolicyServiceInterface;
+    agentService: AgentService;
+    packagePolicyService: PackagePolicyServiceInterface;
+    packageService: PackageService;
+  };
+}
```

Comment:
our route handler context looks pretty much the same as the `CspAppContext`, but we it's easier to consume now.